modulegen/cmd/new/main.go

```diff
@@ -0,0 +1,16 @@
+package new
+
+import (
+	"github.com/spf13/cobra"
+)
+
+func NewCmd() *cobra.Command {
```

Comment:
I think the most accepted pattern for defining commands and subcommands is using the init function: https://github.com/spf13/cobra/blob/main/site/content/user_guide.md#organizing-subcommands

Reply:
I see but it creates a dependency on the parent package while doing this way It's the parent package that knows it's children package

Reply:
I mean within the same `new` package, not in the parent one. The new subcommand would mount the example and module subcommands in their init. For that, we would need a variable for each command.

Said that, we probably don't want a `new` package but a `modules` one, keeping same Go files. 🤔

Reply:
I have change the name of the package. 
Concerning of the use of the init method. This applies only when the command is a var. Those are functions and will be called only one time at the execution of the main() .
Concerning, the use of the init function there are linters that are forbidding it, `gochecknoinits` for example.

Reply:
I think we can live without `gochecknoinits` as we will be using the best-practices defined by Cobra authors. I understand your concern (I also see having multiple inits could lead to code that is difficult to follow) but we will have them very well located. Thanks for raising this concern btw

Reply:
OK ! This shall fit your expectations now ?